### ISSUE TITLE
Got an error on pod install

### ISSUE THREAD

**Issue by MatteoGauthier** [th3rdwave/react-native-safe-area-context#246]
Hey, I firstly installed the npm package `react-native-safe-area-context`, then i run the command `pod install` in the /ios directory. I got this error. Can anyone help me with this issue ?

```
[!] Unable to find a specification for `RCT-Folly` depended upon by `react-native-safe-area-context`

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
```

**Comment by fiznool** [comment]
Did you try running `pod repo update`? You might not have the most up to date version of the podspecs cached locally.

**Comment by MatteoGauthier** [comment]
Here is the result of `pod repo update`

```
➜  ios git:(fix/image-upload) ✗ pod repo update
Updating spec repo `trunk`
➜  ios git:(fix/image-upload) ✗ pod install    
Auto-linking React Native modules for target `Noov`: BVLinearGradient, RNAppleAuthentication, RNCAsyncStorage, RNDeviceInfo, RNGestureHandler, RNLocalize, RNSentry, ReactNativePermissions, appcenter-analytics, appcenter-core, appcenter-crashes, react-native-blur, react-native-fbsdk, react-native-image-picker, react-native-maps, react-native-onesignal, react-native-safe-area-context, react-native-slider, and react-native-webview
Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "react-native-safe-area-context":
  In Podfile:
    react-native-safe-area-context (from `../node_modules/react-native-safe-area-context`)

Specs satisfying the `react-native-safe-area-context (from `../node_modules/react-native-safe-area-context`)` dependency were found, but they required a higher minimum deployment target.
➜  ios git:(fix/image-upload) ✗
```

And my Podfile

```
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
platform :ios, '10.0'
target 'xxxx' do
  config = use_native_modules!
  use_react_native!(:path => config["reactNativePath"])

  # Pods for libraries
  pod 'react-native-geolocation', path: '../node_modules/@react-native-community/geolocation'
  pod 'RNLocalize', :path => '../node_modules/react-native-localize'
  pod 'ReactNativePermissions', :path => '../node_modules/react-native-permissions'

  pod 'react-native-image-picker', :path => '../node_modules/react-native-image-picker'

  target 'xxxx' do
    inherit! :complete
    # Pods for testing
  end
  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable these next few lines.
  # use_flipper!
  # post_install do |installer|
  #   flipper_post_install(installer)
  # end
end
target 'xxxx-tvOS' do
  inherit! :search_paths
  # Pods for testing
  target 'xxxx-tvOSTests' do
    inherit! :complete
    # Pods for testing
  end
end

```

**Comment by fiznool** [comment]
> Specs satisfying the react-native-safe-area-context (from `../node_modules/react-native-safe-area-context`) dependency were found, but they required a higher minimum deployment target.

Your Podfile specifies a minimum iOS target of `10.0` but v4 of `react-native-safe-area-context` requires iOS 11 and above. You can fix this by changing the following line in your Podfile:

``` rb
platform :ios, '10.0'
```

to

```rb
platform :ios, '11.0'
```

**Comment by janicduplessis** [comment]
@fiznool Thanks for helping out, yea that should fix it! Feel free to reopen if you still have an issue.

**Comment by MatteoGauthier** [comment]
It doesn't works for me with 
```rb
platform :ios, '11.0'
```

**Comment by janicduplessis** [comment]
Can you try running `pod update`?

**Comment by MatteoGauthier** [comment]
Same error with `pod update`

**Comment by pedrocardoz0** [comment]
![Captura de Tela 2022-03-02 às 16 06 38](https://user-images.githubusercontent.com/24881317/156430937-f9ac453e-356a-4d82-a1c2-fa65a3b3599c.png)

![Captura de Tela 2022-03-02 às 16 07 07](https://user-images.githubusercontent.com/24881317/156430994-c98c27b4-1bb6-47e1-b71f-dc556d624e0a.png)

I'm facing the same issue

**Comment by pedrocardoz0** [comment]
Try to use any other version I switched to `3.3.2`.

**Comment by fiznool** [comment]
Sounds like an old dependency is cached. Try removing your Podfile.lock and run `pod install` again.

**Comment by MatteoGauthier** [comment]
> Sounds like an old dependency is cached. Try removing your Podfile.lock and run `pod install` again.

Didn't works too 😕

**Comment by rafabulsing** [comment]
Same error here. `platform :ios` is set as `12.0`, and the version of `react-native` is `0.63.4`. Have tried using `pod repo update` and deleting `Podfile.lock`, but `pod install` still errors out. Any ideas?

**Comment by janicduplessis** [comment]
What version of cocoapods are you using? Maybe updating to the latest version would fix it?

**Comment by MatteoGauthier** [comment]
> What version of cocoapods are you using? Maybe updating to the latest version would fix it?

I have the 1.11.2 version of cocoapods (latest)

**Comment by fiznool** [comment]
Another option is to try using [react-native-clean-project](https://www.npmjs.com/package/react-native-clean-project) to reset all cached dependencies. Then try running `npm install && cd ios && pod install` again.

**Comment by jspizziri** [comment]
# TL;DR;

If you're not running a modern version of `react-native` you're going to get this error. And you can always check `node_modules/react-native/scripts/react_native_pods.rb` to see if its specified in your version. Here's a primitive check:

```
cat node_modules/react-native/scripts/react_native_pods.rb | grep "/third-party-podspecs/RCT-Folly.podspec"
```

CC @rafabulsing your version of `react-native` just isn't going to work.

# Detail

Alright, I was having this issue too. My usecase is that I'm building a `react-native` library for which [I generated the scaffolding via `npx create-react-native-library` as documented here](https://reactnative.dev/docs/native-modules-setup).

I was then trying to install `react-native-safe-area-context` in the example app I was getting the same error documented above. So I started doing some digging, and realized that the generated app had [`react-native@0.63.4`](https://github.com/facebook/react-native/releases/tag/v0.63.4) installed which is quite old. I think the technical term is "super-duper old"... and that version doesn't have the [`third-party-podspecs` library installed](https://github.com/facebook/react-native/blame/main/scripts/react_native_pods.rb#L80).

In my case it looks like the issue is with `npx create-react-native-library`, not this lib.

**Comment by JongtaekChoi** [comment]
@jspizziri  Thanks. I changed the react-native version and it works!

**Comment by MatteoGauthier** [comment]
Thanks 😉

**Comment by rafabulsing** [comment]
Thank you guys so much for the help, and for documenting the issue on the README. (And sorry for the delayed response!)
